### PR TITLE
support date_add date_sub timestamp_add bulitin function

### PR DESCRIPTION
**Pull request by bbbearxyz**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2622

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/container/types/interval.go [hunk 1]
```diff
@@ -202,3 +202,91 @@ func NormalizeInterval(s string, it IntervalType) (ret int64, rettype IntervalTy
 	}
 	return
 }
+
+func ParseIntervalToList(s string, it IntervalType, isDateAdd bool) (rets []int32, rettypes []IntervalType, err error) {
+	rets, err = parseInts(s)
+	// we need to solve the problem XXX_MicroSecond
```
Comment (fengttt) [inline pkg/container/types/interval.go:208]: Are you sure this is the right behaviour?     Please take a look at parseInt function.    Why do you need this function again?

At least, if you are sure what your function is correct, fix the parseInt function.

Comment (bbbearxyz) [reply pkg/container/types/interval.go:208]: parseInts just parse "1 1:1:1:1.1" to slice[1, 1, 1, 1, 1, 1], I think I am right. why do you think I am wrong, please tell me the detail

Comment (bbbearxyz) [reply pkg/container/types/interval.go:208]: @fengttt  when I have "1.1" SECOND_MICROSECOND, the microsecond should be 100000 instead of 1.

Comment (bbbearxyz) [reply pkg/container/types/interval.go:208]: @fengttt do you think I will fix the parseint function or fix the NormalizeInterval function

**Review by yingfeng — CHANGES_REQUESTED** [review]
Fix static check

**Pull request merged**